Source/Core/Common/Hash.cpp

```diff
@@ -530,4 +532,13 @@ void SetHash64Function()
     ptrHashFunction = &GetMurmurHash3;
   }
 }
+
+std::string MakeGameID(std::string file_name)
+{
+  size_t lastdot = file_name.find_last_of(".");
+  if (lastdot == std::string::npos)
+    return "ID(" + file_name + ")";
+  return "ID(" + file_name.substr(0, lastdot) + ")";
```

Comment:
Why not just return `file_name` or `file_name.substr(0, lastdot)` directly? In my opinion, the `ID(...)` looks somewhat ugly and on Linux it also makes for ugly paths and INI file names that require the user to quote them.